packages/dds/tree/src/simple-tree/viewSchemaToSimpleSchema.ts

```diff
@@ -0,0 +1,267 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert, unreachableCase } from "@fluidframework/core-utils/internal";
+import {
+	FieldKind,
+	NodeKind,
+	normalizeFieldSchema,
+	type FieldSchema,
+	type ImplicitAllowedTypes,
+	type TreeNodeSchema,
+} from "./schemaTypes.js";
+import type {
+	SimpleArrayNodeSchema,
+	SimpleFieldSchema,
+	SimpleFieldSchemaKind,
+	SimpleLeafNodeSchema,
+	SimpleLeafSchemaKind,
+	SimpleMapNodeSchema,
+	SimpleNodeSchema,
+	SimpleObjectNodeSchema,
+	SimpleTreeSchema,
+} from "./simpleSchema.js";
+import { ValueSchema } from "../core/index.js";
+import { fail } from "../util/index.js";
+import { isObjectNodeSchema, type ObjectNodeSchema } from "./objectNodeTypes.js";
+
+/**
+ * Converts a "view" schema to a "simple" schema representation.
+ */
+export function toSimpleTreeSchema(schema: ImplicitAllowedTypes): SimpleTreeSchema {
```

Comment:
I really liked your example output for the toJsonSchema method docs. Would be nice in the tsdoc here as well.

Reply:
Also, What is the purpose of the SimpleSchema intermediary? Is there a use case for this other than being converted to JsonSchema?

Reply:
Yeah, it's better suited for programmatic usage generally. There is some vague interest in exposing something like it in the future. Not directly applicable in this PR, but will almost certainly be exposed (in some form) in the near future.

I plan to use it in the devtools visualizer for SharedTree, as one example.

Reply:
Example added 🙂